### DISCUSSION TITLE
Can we use useServerSeoMeta? Or is UseSeoMeta fine for SEO?

### DISCUSSION THREAD

**Discussion by notflip** [nuxt-modules/i18n#2713] [Q&A]
I'm using a headless CMS to load all of my date (pages, slug, navigation). I'm using `setI18nParams` to set the correct slugs for my navigation, the only thing that I don't yet understand is the `useSeoMeta` and `useServerSeoMeta`

I've used `useServerSeoMeta` for all my previous websites (that's SSR, right?) and now when switching the language I see the meta data is not updating, I suppose that's because switching the language happens on the frontend, after the meta has been loading with `useServerSeoMeta`. 

My question is, is useSeoMeta equally fine for search engines?

**Comment by gaomingzhao666** [comment]
Hello (●'◡'●).

Use `useServerSeoMeta` if you do not want to return any seo data to the client, which means that the seo metadata are initialized in the server, use it when you have a static data website such as portfolio or using a CMS.

Use `useSeoMeta` if you want to make the seo metadata reactive, which means that those metadata are stored in client-side dynamically, 
 use it when you have a dynamic website such as an online shop, imagine that the goods are reactive to show to search engine, each goods have a unique title and description to dynamically show to search engine like `Google.com`

and `useHead` is same to `useSeoMeta` but it's have no `flat support`, which means it can not help you to avoid typos and common mistakes.
